File: Scripts/results/generated_code_results/RN_gpt/Wellness/Headspace/ExploreScreen.js
import React from 'react';
import { View, Text, TextInput, TouchableOpacity, Image, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native'; // Add this import for navigation

const ExploreScreen = () => {
  const navigation = useNavigation(); // Initialize useNavigation hook

  return (
    <View style={styles.container}>
      <TextInput 
        style={styles.searchBar} 
        placeholder="Search exercises"
        placeholderTextColor="#ccc"
      />
      
      <View style={styles.row}>
        <TouchableOpacity style={[styles.categoryButton, styles.meditate]}>
          <Text style={styles.categoryText}>Meditate</Text>
        </TouchableOpacity>
        <TouchableOpacity style={[styles.categoryButton, styles.sleep]}>
          <Text style={styles.categoryText}>Sleep</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.row}>
        <TouchableOpacity 
          style={[styles.categoryButton, styles.move]}
          onPress={() => navigation.navigate('Move')} // Redirection to Move
        >
          <Text style={styles.categoryText}>Move</Text>
        </TouchableOpacity>
        <TouchableOpacity style={[styles.categoryButton, styles.music]}>
          <Text style={styles.categoryText}>Music</Text>
        </TouchableOpacity>
      </View>
      
      <TouchableOpacity style={[styles.categoryButton, styles.podcasts]}>
        <Text style={styles.categoryText}>Podcasts</Text>
      </TouchableOpacity>

      <View style={styles.promoSection}>
        <Text style={styles.promoText}>Unlock the Headspace Library</Text>
        <TouchableOpacity style={styles.promoButton}>
          <Text style={styles.promoButtonText}>Start My Free Trial</Text>
        </TouchableOpacity>
      </View>

      <TouchableOpacity style={[styles.featuredContent, styles.politics]}>
        <Text style={styles.featuredText}>Politics Without Panic</Text>
      </TouchableOpacity>

      <TouchableOpacity style={[styles.featuredContent, styles.anxiety]}>
        <Text style={styles.featuredText}>Calming Everyday Anxiety</Text>
      </TouchableOpacity>

      <View style={styles.bottomNavBar}>
        <TouchableOpacity 
          style={styles.navItem}
          onPress={() => navigation.navigate('Home')} // Redirection to Home
        >
          <Image source={require('../assets/snack-icon.png')} style={styles.icon} />
          <Text style={styles.navText}>Today</Text>
        </TouchableOpacity>
        <View style={styles.navItemActive}>
          <Image source={require('../assets/snack-icon.png')} style={styles.icon} />
          <Text style={styles.navTextActive}>Explore</Text>
        </View>
        <TouchableOpacity 
          style={styles.navItem}
          onPress={() => navigation.navigate('CalmMethods')} // Redirection to CalmMethods
        >
          <Image source={require('../assets/snack-icon.png')} style={styles.icon} />
          <Text style={styles.navText}>Profile</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#0e1222',
    padding: 16,
  },
  searchBar: {
    backgroundColor: '#1c2036',
    borderRadius: 8,
    padding: 8,
    color: '#fff',
    marginBottom: 16,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 16,
  },
  categoryButton: {
    flex: 1,
    margin: 4,
    borderRadius: 8,
    padding: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  categoryText: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 16,
  },
  meditate: {
    backgroundColor: 'orange',
  },
  sleep: {
    backgroundColor: 'purple',
  },
  move: {
    backgroundColor: 'pink',
  },
  music: {
    backgroundColor: 'blue',
  },
  podcasts: {
    backgroundColor: 'yellow',
    paddingVertical: 30,
  },
  promoSection: {
    backgroundColor: '#1c2036',
    borderRadius: 8,
    padding: 20,
    alignItems: 'center',
    marginBottom: 16,
  },
  promoText: {
    color: '#fff',
    fontWeight: 'bold',
    marginBottom: 10,
  },
  promoButton: {
    backgroundColor: 'blue',
    borderRadius: 8,
    padding: 10,
    alignItems: 'center',
  },
  promoButtonText: {
    color: '#fff',
    fontWeight: 'bold',
  },
  featuredContent: {
    borderRadius: 8,
    padding: 20,
    alignItems: 'center',
    marginBottom: 16,
  },
  politics: {
    backgroundColor: 'pink',
  },
  anxiety: {
    backgroundColor: 'blue',
  },
  featuredText: {
    color: '#fff',
    fontWeight: 'bold',
  },
  bottomNavBar: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: '#1c2036',
    padding: 20,
  },
  navItem: {
    alignItems: 'center',
  },
  navItemActive: {
    alignItems: 'center',
  },
  icon: {
    width: 24,
    height: 24,
    marginBottom: 4,
  },
  navText: {
    color: '#ccc',
  },
  navTextActive: {
    color: '#fff',
  },
});

export default ExploreScreen;

File: Scripts/results/generated_code_results/Flutter/Wellness/Headspace/ExploreScreen.dart
import 'package:flutter/material.dart';
class ExploreScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
body: SafeArea(
child: Column(
children: [
Padding(
padding: const EdgeInsets.all(16.0),
child: TextField(
decoration: InputDecoration(
hintText: 'Search exercises',
prefixIcon: Icon(Icons.search),
border: OutlineInputBorder(
borderRadius: BorderRadius.circular(30),
),
),
),
),
Expanded(
child: SingleChildScrollView(
child: Column(
children: [
GridView.count(
shrinkWrap: true,
physics: NeverScrollableScrollPhysics(),
crossAxisCount: 2,
childAspectRatio: 2.5,
padding: EdgeInsets.all(16),
mainAxisSpacing: 16,
crossAxisSpacing: 16,
children: [
_buildCategoryButton('Meditate', Colors.orange, context),
_buildCategoryButton('Sleep', Colors.purple, context),
_buildCategoryButton('Move', Colors.pink, context),
_buildCategoryButton('Music', Colors.blue, context),
],
),
_buildPodcastsButton(),
_buildUnlockLibrarySection(context),
_buildFeaturedContent(context),
],
),
),
),
],
),
),
bottomNavigationBar: BottomNavigationBar(
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Today'),
BottomNavigationBarItem(icon: Icon(Icons.explore), label: 'Explore'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: '子骆'),
],
currentIndex: 1,
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Home');
} else if (index == 2) {
Navigator.pushNamed(context, 'Profile');
}
},
),
);
}
Widget _buildCategoryButton(String title, Color color, BuildContext context) {
return ElevatedButton(
style: ElevatedButton.styleFrom(
backgroundColor: color,
shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(15)),
),
onPressed: () {
if (title == 'Move') {
Navigator.pushNamed(context, 'Move');
}
},
child: Text(title, style: TextStyle(fontSize: 18, color: Colors.white)),
);
}
Widget _buildPodcastsButton() {
return Container(
margin: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
child: ElevatedButton(
style: ElevatedButton.styleFrom(
backgroundColor: Colors.yellow,
shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(15)),
),
onPressed: () {},
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text('Podcasts', style: TextStyle(fontSize: 18, color: Colors.black)),
Image.asset('assets/images/template.png', height: 40),
],
),
),
);
}
Widget _buildUnlockLibrarySection(BuildContext context) {
return Container(
margin: EdgeInsets.all(16),
padding: EdgeInsets.all(16),
decoration: BoxDecoration(
color: Colors.grey[800],
borderRadius: BorderRadius.circular(15),
),
child: Column(
children: [
Text(
'Unlock the Headspace Library',
style: TextStyle(color: Colors.white, fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
ElevatedButton(
style: ElevatedButton.styleFrom(
backgroundColor: Colors.blue,
shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(30)),
),
onPressed: () {},
child: Text('Start My Free Trial', style: TextStyle(fontSize: 16)),
),
],
),
);
}
Widget _buildFeaturedContent(BuildContext context) {
return Column(
children: [
_buildFeaturedCard('Politics Without Panic', Colors.pink),
_buildFeaturedCard('Calming Everyday Anxiety', Colors.blue, onTap: () {
Navigator.pushNamed(context, 'CalmMethods');
}),
],
);
}
Widget _buildFeaturedCard(String title, Color color, {VoidCallback? onTap}) {
return GestureDetector(
onTap: onTap,
child: Container(
margin: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
padding: EdgeInsets.all(16),
decoration: BoxDecoration(
color: color,
borderRadius: BorderRadius.circular(15),
),
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text(title, style: TextStyle(color: Colors.white, fontSize: 18, fontWeight: FontWeight.bold)),
Image.asset('assets/images/template.png', height: 40),
],
),
),
);
}
}